### PR TITLE
(EAI-457) [UI] Custom chatbot trigger helpers

### PR DESCRIPTION
**Pull request by nlarew**
Jira: (EAI-457) [UI] Custom chatbot trigger helpers

## Changes

- Refactors most of `InputBarTrigger` data and logic into a separate, re-usable hook
- Cleans up unnecessary `classnames` dependency in favor of Emotion's basically equivalent `cx`

## Notes

- Makes it easier for devs to make input bar triggers like ours but with custom components/logic
- Most applicable use is for upcoming DOP work on a new trigger

### CODE REVIEW

File: packages/mongodb-chatbot-ui/src/FloatingActionButtonTrigger.tsx [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import { css } from "@emotion/css";
-import classNames from "classnames";
+import { css, cx } from "@emotion/css";
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/FloatingActionButtonTrigger.tsx:1]: why these changes? can you note why in the PR description?

File: packages/mongodb-chatbot-ui/src/useTextInputTrigger.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+export type UseTextInputTriggerArgs = {
+  placeholder?: string;
+  fatalErrorMessage?: string;
+};
+
+/**
+  The base props for a Chatbot trigger component that allows the user to input text.
+ */
+export type ChatbotTextInputTriggerProps = ChatbotTriggerProps &
```
Comment (mongodben) [inline packages/mongodb-chatbot-ui/src/useTextInputTrigger.ts:15]: do we need to update the public facing docs for this PR?

File: packages/mongodb-chatbot-ui/src/FloatingActionButtonTrigger.tsx (hunk 1 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/src/FloatingActionButtonTrigger.tsx:1]: Basically just tidying up - we don't need this lib because emotion has their own version baked in

File: packages/mongodb-chatbot-ui/src/useTextInputTrigger.ts (hunk 2 above)
Comment (nlarew) [reply packages/mongodb-chatbot-ui/src/useTextInputTrigger.ts:15]: Nothing required, though we could cover custom triggers/view more generally in the docs & mention this there.

https://jira.mongodb.org/browse/EAI-468
